v2/datastream-to-spanner/src/main/java/com/google/cloud/teleport/v2/templates/SpannerTransactionWriterDoFn.java

```diff
@@ -102,6 +105,11 @@ class SpannerTransactionWriterDoFn extends DoFn<FailsafeElement<String, String>,
   /* The run mode, whether it is regular or retry. */
   private final Boolean isRegularRunMode;
 
+  private transient AtomicLong transactionAttemptCount;
+  private transient AtomicBoolean isInTransaction;
+  private transient AtomicBoolean keepWatchdogRunning;
+  private transient Thread watchdogThread;
```

Comment:
Add a detailed doc on what does the WatchDog and why is it needed.

Reply:
done